### PR TITLE
Feature/show2 1049 home feed

### PR DESCRIPTION
**Pull request by alantoa**
# Why
- we want to get rid of `recyclerlistview` on the web, and `VirtuosoList` is better on the web.
- `useInfiniteListQuerySWR` hooks' load more has a bug, we seem to haven't noticed.
- notifications list loads-more have bug.

# How

- improve `VirtuosoList`, on prop `numColumns` === 1. use `Virtuoso` component, on prop `numColumns` > 1. we use `VirtuosoGrid` component. will be better for performance and features. 

- add `isReachingEnd` field on `useInfiniteListQuerySWR` hooks, and the `fetchMore` method will not be running when the list is loading completed.
- replace `FlatList` of in `notifications` with new `InfiniteScrollList`.

# InfiniteScrollList On Native Progress ⌛️
- will try to replace `FlatList` in `infinite-scroll-list` with `FlashList`, I need to do some tests before this, make sure it work well!

# Test Plan

- Test if the new Home/Trending list performs well.
- Check if the notification list looks good!

### CODE REVIEW

File: packages/app/components/feed/feed.md.tsx [hunk 1]
```diff
@@ -162,109 +155,69 @@ export const FeedList = () => {
...
-  const { width: screenWidth, height } = useWindowDimensions();
-  let dataProvider = useMemo(
-    () =>
-      new DataProvider((r1, r2) => {
-        return r1.nft_id !== r2.nft_id;
-      }).cloneWithRows(data),
-    [data]
-  );
-
-  const _layoutProvider = useMemo(
```
Comment (intergalacticspacehighway) [inline packages/app/components/feed/feed.md.tsx:187]: mah man. This API used to give me PTSD 😌

File: packages/app/components/trending/trending-creators-list.md.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { useState, useCallback } from "react";
+import { ListRenderItemInfo } from "react-native";
+
```
Comment (intergalacticspacehighway) [inline packages/app/components/trending/trending-creators-list.md.tsx:3]: no problem with this file but I think we won't be using trending creators for the time being.

File: packages/app/components/feed/feed.md.tsx (hunk 1 above)
Comment (alantoa) [reply packages/app/components/feed/feed.md.tsx:187]: +1!! especially on the web, too many weird issues!
I'm very empathetic and battle with this API! 🤣

File: packages/app/global.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+declare module "@biconomy/mexa";
```
Comment (intergalacticspacehighway) [inline packages/app/global.d.ts:1]: 💯

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM 🔥 🔥 🔥

File: packages/app/components/trending/trending-creators-list.md.tsx (hunk 2 above)
Comment (alantoa) [inline packages/app/components/trending/trending-creators-list.md.tsx:3]: yeah I know, I have been tangled for a while to change this file 🤣, but I thought I might use it in the future, so I improved. (maybe it's my OTD I want consistency.)

File: packages/app/global.d.ts (hunk 3 above)
Comment (axeldelafosse) [reply packages/app/global.d.ts:1]: Unrelated but we can probably get rid of this package altogether

File: packages/app/providers/user-provider.tsx [hunk 4]
```diff
@@ -60,7 +60,7 @@ export function UserProvider({ children }: UserProviderProps) {
     const identifyAndRegisterPushNotification = async () => {
       if (data) {
         // Identify user
-        mixpanel.identify(data.data.profile.profile_id.toString());
+        mixpanel.identify(data.data?.profile.profile_id.toString());
```
Comment (axeldelafosse) [inline packages/app/providers/user-provider.tsx:63]: We can remove this `mixpanel.identify` -- we don't use it anymore

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

**Pull request merged**